### ISSUE TITLE
[Graph] Graph view need to support Branch nodes

### ISSUE THREAD

**Issue by anrusina** [flyteorg/flyteconsole#377]
### Issue 

"Currently branch node view is messed up and hard to follow" - per Ketan statement.
Examples: 
`control_flow.run_merge_sort.merge_sort` - only failed items are rendered.
https://demo.nuclyde.io/console/projects/flytesnacks/domains/development/executions/f1c3d1514116a44019fe?duration=all
`versions view`
https://demo.nuclyde.io/console/projects/flytesnacks/domains/development/workflows/core.control_flow.run_merge_sort.merge_sort/version/ARW5epeOV80feOjLTUXmtQ==
`dolt.dolt_branch_example.wf`
https://demo.nuclyde.io/console/projects/flytesnacks/domains/development/executions/arnb8thjdrpqqmsgqrmc?duration=all

### Solution

At this point we haven't treated the branch node in any way that is different from regular python_task in DagV2, as such we don't have enough information to show it in a proper way

This work should be approached in few steps:
[ ] Understand Branch node type (spike)
[ ] Design requirements for how we will show branch node in Graph View
[ ] Add Branch node to DagV2 info (?)

**Comment by cosmicBboy** [comment]
won't fix
